ui/register_react_component.md

```diff
@@ -0,0 +1,107 @@
+### Register react component
+In ManageIQ project there is a way of sharing components, getting their state and control their flow by something which is called ManageIQ component.
+
+This registry and usage consists of three steps, define blueprint that holds the react component with its props. Then you will define it inside shared components (use blueprint created in step 1). And when someone decides to use such component he will call function newInstance with custom props and element to which such component is bound to.
+
+This way using such component is abstract for user, he doesn't know if such component was written in pure JS, React, Angular or whatever.
+
+#### 1. Define react component
+This is your first step of creating shared abstract component. You may use our custom function which fits react, should you choose angular, Vue.js or whatever you'll have to write your own funcion.
+
+When using react blueprint function, just wrap your component (or HTML) with `addReact` function and as argument use callback which will be called with props. These props will be passed when new isntance of this component is created.
+
+```JSX
+import React from 'react';
+import addReact from 'helpers';
+import { YourComponent } from 'your-components/custom-component';
+
+addReact('YourComponentName', YourComponent);
+```
+
+#### 2. New instance
+When you want to create new instance of recently created component just call `newInstance` function with these arguments
+* name - name of the component you wish to use.
+* props - object with props that will be mapped to your components (used when creating blueprint)
+* element - DOM element where this new instance of component should be rendered.
+```JS
+import componentFactory from 'helpers';
```

Comment:
`componentFactory` is not a default export, should be `import { componentFactory } from ...`